<launch>
  <param name="robot/type" value="pepper" />
  <param name="robot/name" command='bash -c "getent hosts $NAO_IP | tr -s \" \" | cut -d\  -f 2 | cut -d. -f 1"' />
  <arg name="nao_ip" value="$(env NAO_IP)"/>
  <arg name="roscore_ip" value="$(env ROS_IP)"/>
  <arg name="network_interface"   default="eth0" />
  <arg name="launch_twitter" default="false" />
  <arg name="launch_joy" default="true" />
  <arg name="launch_play_audio_stream" default="true" />
  <arg name="launch_dashboard" default="true" />
  <arg name="autonomous_wakeup" default="false" />
  <include file="$(find pepper_bringup)/launch/pepper_full.launch" >
    <arg name="nao_ip" value="$(arg nao_ip)"/>
    <arg name="roscore_ip" value="$(arg roscore_ip)"/>
    <arg name="network_interface" value="$(arg network_interface)"/>
  </include>

  <node pkg="jsk_pepper_startup" name="take_wakeup_pose" type="take_wakeup_pose.py" if="$(arg autonomous_wakeup)" />

  <!-- <include file="$(find turtlebot_teleop)launch/keyboard_teleop.launch" /> -->
  <node pkg="joy" type="joy_node" name="joy_node" respawn="true" if="$(arg launch_joy)">
    <param name="dev" value="/dev/input/js0" />
    <param name="deadzone" value="0.3" />
    <param name="autorepeat_rate" value="20" />
  </node>
  <node pkg="teleop_twist_joy" name="teleop_twist_joy"
        type="teleop_node" respawn="true" if="$(arg launch_joy)">
    <rosparam>
# stop button: Left trigger button
axis_linear:
  x: 1.0
  y: 0.0
scale_linear:
  x: 0.2
  y: 0.2
scale_linear_turbo:
  x: 0.5
  y: 0.5

axis_angular:
  yaw: 3
scale_angular:
  yaw: 1.0
scale_augular_turbo:
  yaw: 1.5
  yaw: 1.5

enable_button: 4  # Left trigger button
enable_turbo_button: 0  # A button
    </rosparam>
  </node>
  <group if="$(arg launch_twitter)">
    <node pkg="jsk_pepper_startup" name="twitter_server" type="tweet-server.l" respawn="true" />
    <node pkg="rostwitter" name="tweet_node" type="tweet.py" respawn="true"/>
    <node pkg="image_view" name="image_saver" type="image_saver" respawn="true" >
      <remap from="image" to="/pepper_robot/camera/front/image_raw" />
      <param name="save_all_image" value="false" />
      <param name="filename_format" value="/tmp/pepper_camera.png" />
    </node>
  </group>

  <!-- If use ROS >= kinetic, please clone https://github.com/kochigami/naoqi_dashboard/tree/kochigami-develop (This includes https://github.com/ros-naoqi/naoqi_dashboard/pull/3) -->
  <include file="$(find naoqi_dashboard)/launch/naoqi_dashboard.launch" if="$(arg launch_dashboard)" />

  <node pkg="jsk_pepper_startup" name="joy_client" type="joy-client.l" respawn="true" if="$(arg launch_joy)" />

  <node pkg="jsk_pepper_startup" name="play_audio_stream" type="play_audio_stream.py" respawn="true" if="$(arg launch_play_audio_stream)" />

  <node pkg="jsk_pepper_startup" name="list_up_ros_node" type="list_up_ros_node.py" respawn="true" />
  
  <node name="speaking_program_is_started_or_terminated"
        pkg="roseus" type="roseus"
        args="$(find jsk_robot_startup)/lifelog/speaking-program-is-started-or-terminated.l &quot;(progn (unix:sleep 3) (good-morning) (ros::spin))&quot;" respawn="true" />
</launch>
